src/core/src/main/scala/play/api/db/slick/DatabaseConfigProvider.scala

```diff
@@ -158,7 +158,7 @@ trait HasDatabaseConfig[P <: BasicProfile] {
   @deprecated("Use `profile` instead of `driver`", "2.1")
   protected final lazy val driver: P = dbConfig.profile // field is lazy to avoid early initializer problems.
   /** The Slick database extracted from `dbConfig`. */
-  protected final def db: P#Backend#Database = dbConfig.db
```

Comment:
I understood that this is Path Dependent Types but I didn't find a way to make it work. Any insight here would be much appreciated!